ivy/functional/frontends/numpy/creation_routines/numerical_ranges.py

```diff
@@ -156,3 +158,14 @@ def __init__(self):
 
 
 ogrid = OGrid()
+
+
+@handle_numpy_dtype
+@to_ivy_arrays_and_back
+def geomspace(start, stop, num=50, endpoint=True, dtype=None, axis=0):
+    start, stop = promote_types_of_numpy_inputs(start, stop)
+    cr = ivy.log((stop / start), where=start != 0) / (num - 1 if endpoint else num)
+    x = ivy.linspace(0, cr * (num - 1 if endpoint else num), num, endpoint=endpoint, dtype=dtype, axis=axis)
+    x = ivy.exp(x)
+    x = start * x
+    return x
```

Comment:
It seems like numpy has some logic which ensures the start and end elemnets are exactly the start and end elementes specified, you can add this by including the following 3 lines:
```
x[0] = start
if endpoint:
    x[-1] = stop
```